src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -57,8 +65,37 @@ async Task<CompletableSynchronizedStorageSession> AdaptOrOpenNewSynchronizedStor
                    ?? await synchronizedStorage.OpenSession(contextBag).ConfigureAwait(false);
         }
 
+        static void LogHandlersInvocation(IIncomingLogicalMessageContext context, List<MessageHandler> handlersToInvoke)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"Processing message type: {context.Message.MessageType}");
+            builder.AppendLine("Message headers: ");
+
+            foreach (var kvp in context.Headers)
+            {
+                builder.Append($"{kvp.Key}:{kvp.Value}, ");
+            }
+
+            builder.Length -= 2;
+
+            builder.AppendLine();
+            builder.Append("Handlers to invoke: ");
+
+            foreach (var messageHandler in handlersToInvoke)
+            {
+                builder.Append($"{messageHandler.HandlerType.FullName} => ");
```

Comment:
Same here. Why not just have each handler to invoke on it's own line?

Reply:
If https://github.com/Particular/NServiceBus/pull/5230#discussion_r210667449 is a go, make sense.